core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -79,6 +79,7 @@
     public static final String SCALE_WRITERS = "scale_writers";
     public static final String TASK_SCALE_WRITERS_ENABLED = "task_scale_writers_enabled";
     public static final String TASK_SCALE_WRITERS_MAX_WRITER_COUNT = "task_scale_writers_max_writer_count";
+    public static final String TASK_SCALE_WRITERS_PARTITION_COUNT = "task_scale_writers_partition_count";
```

Comment:
I think it's very technical. Let's just keep it as config (maybe) or just some high enough number (like 10k).